src/main/java/org/cyverse/cas/ldap/rfc2307/config/Rfc2307LdapAuthenticationConfiguration.java

```diff
@@ -46,135 +45,120 @@
+                final LdapAuthenticationHandler handler = new Rfc2307LdapAuthenticationHandler(l.getName(),
+                    servicesManager, ldapPrincipalFactory(), l.getOrder(), authenticator, strategy, groupMembershipResolver);
+                handler.setCollectDnAttribute(l.isCollectDnAttribute());
+
+                final List<String> additionalAttributes = l.getAdditionalAttributes();
+                if (StringUtils.isNotBlank(l.getPrincipalAttributeId())) {
+                    additionalAttributes.add(l.getPrincipalAttributeId());
+                }
+                if (StringUtils.isNotBlank(l.getPrincipalDnAttributeName())) {
+                    handler.setPrincipalDnAttributeName(l.getPrincipalDnAttributeName());
+                }
+                handler.setAllowMultiplePrincipalAttributeValues(l.isAllowMultiplePrincipalAttributeValues());
+                handler.setAllowMissingPrincipalAttributeValue(l.isAllowMissingPrincipalAttributeValue());
+                handler.setPasswordEncoder(PasswordEncoderUtils.newPasswordEncoder(l.getPasswordEncoder()));
+                handler.setPrincipalNameTransformer(PrincipalNameTransformerUtils.newPrincipalNameTransformer(l.getPrincipalTransformation()));
+
+                if (StringUtils.isNotBlank(l.getCredentialCriteria())) {
+                    LOGGER.debug("Ldap authentication for [{}] is filtering credentials by [{}]",
+                        l.getLdapUrl(), l.getCredentialCriteria());
+                    handler.setCredentialSelectionPredicate(CoreAuthenticationUtils.newCredentialSelectionPredicate(l.getCredentialCriteria()));
+                }
+
+                if (StringUtils.isBlank(l.getPrincipalAttributeId())) {
+                    LOGGER.debug("No principal id attribute is found for LDAP authentication via [{}]", l.getLdapUrl());
+                } else {
+                    handler.setPrincipalIdAttribute(l.getPrincipalAttributeId());
+                    LOGGER.debug("Using principal id attribute [{}] for LDAP authentication via [{}]", l.getPrincipalAttributeId(), l.getLdapUrl());
+                }
+
+                final LdapPasswordPolicyProperties passwordPolicy = l.getPasswordPolicy();
+                if (passwordPolicy.isEnabled()) {
+                    LOGGER.debug("Password policy is enabled for [{}]. Constructing password policy configuration", l.getLdapUrl());
+                    final PasswordPolicyConfiguration cfg = createLdapPasswordPolicyConfiguration(passwordPolicy, authenticator, multiMapAttributes);
+                    handler.setPasswordPolicyConfiguration(cfg);
+                }
+
+                final Map<String, Object> attributes = CollectionUtils.wrap(multiMapAttributes);
+                handler.setPrincipalAttributeMap(attributes);
+
+                LOGGER.debug("Initializing LDAP authentication handler for [{}]", l.getLdapUrl());
+                handler.initialize();
+                handlers.add(handler);
+            });
```

Comment:
Last time, I refactored this chunk of code into a separate method. I didn't bother doing that this time.